File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 07/04/2026, 11:43:56
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> mp;
        
        

        int ans = 0 ; 
        int n = s.size();
        int l = 0; 
        int count = 0 ;
        if ( n == 0) return 0 ; 
        

        for ( int j = 0 ; j<n;j++)
        {
            if (  mp.find(s[j]) != mp.end() && l <=mp[s[j]]){
                l = mp[s[j]]+1;
            }
         
        
            count = j-l+1;
            ans = max(ans,count);
            mp[s[j]] = j ;
            
        }
        return ans;
    }
};

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.js
// Last updated: 07/04/2026, 11:44:01
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {

    let list = new Map();
    let output = 0 ;
    let pos = -1 ; 

        for (let i = 0 ; i<s.length;i++){
            if (list.has(s[i]) && list.get(s[i]) > pos){
                pos= list.get(s[i]);

            }
            list.set(s[i],i);
            output = Math.max(output,i-pos);
        }
        return output;

};